Source/AliveLibAO/Bullet.cpp

```diff
@@ -35,4 +43,202 @@ BaseGameObject* Bullet::VDestructor(signed int flags)
+                    if (Math_RandomRange_450F20(0, 128) < 64)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    SFX_Play_43AD70(SoundEffect::Unknown_2, volume, 0);
+                    field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                    return;
+                }
+            }
+            else
+            {
+                if (!pShotObj)
+                {
+                    if ((signed __int16) Math_RandomRange_450F20(0, 100) >= 70)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    if (Math_RandomRange_450F20(0, 128) < 64)
+                    {
+                        SFX_Play_43AD70(SoundEffect::Unknown_1, volume, 0);
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    SFX_Play_43AD70(SoundEffect::Unknown_2, volume, 0);
+                    field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                    return;
+                }
+            }
+            if (pShotObj->VTakeDamage_401920(this))
+            {
+                SFX_Play_43AE60(SoundEffect::AirStream_28, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, 2000, 0);
+                SFX_Play_43AE60(SoundEffect::MeatBounce_43, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, Math_RandomRange_450F20(300, 700), 0);
+                SFX_Play_43AE60(SoundEffect::KillEffect_64, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, Math_RandomRange_450F20(900, 1400), 0);
+            }
+            field_6_flags.Set(BaseGameObject::eDead_Bit3);
+        }
+        case BulletType::Type_2:
+        {
+            FP v5;
+            FP v7;
+            FP hitX;
+            FP hitY;
+            PSX_RECT shootRect = {};
+            if (sControlledCharacter_50767C == (BaseAliveGameObject *) gElum_507680)
+            {
+                v5 = gElum_507680->field_B4_velx * FP_FromInteger(4);
+                v7 = sActiveHero_507678->field_A8_xpos;
+            }
+            else
+            {
+                v5 = sActiveHero_507678->field_B4_velx * FP_FromInteger(2);
+                if (field_18_xpos >= sActiveHero_507678->field_A8_xpos)
+                {
+                    v7 = sActiveHero_507678->field_A8_xpos + FP_FromInteger(field_34 << 4);
+                }
+                else
+                {
+                    v7 = sActiveHero_507678->field_A8_xpos - FP_FromInteger(field_34 << 4);
+                }
+            }
+            shootRect.x = FP_GetExponent(v7 - v5);
+            shootRect.w = shootRect.x + 2;
+            shootRect.y = FP_GetExponent(sActiveHero_507678->field_AC_ypos)
+                + sActiveHero_507678->field_10_anim.Get_FrameHeader_403A00(-1)->field_8_data.points[2].y //or points 3?!
```

Comment:
I'm not sure this is correct, what does AE code do here? I think its the max y point of the bounding rect?